### ISSUE TITLE
Compatibility renderer 3D MSAA suspected memory leak (Android)

### ISSUE THREAD

**Issue by Bimbam360** [godotengine/godot#97967]
### Tested versions

v4.4.dev3.official [f4af8201b]

### System information

OpenGL API OpenGL ES 3.2 v1.r32p1-01eac0.394145956bc7cd8e697b330aba11e3d3 - Compatibility - Using Device: ARM - Mali-G57 MC2

### Issue description

When enabling MSAA 3D on Android the 'Native' memory use in the android profiler continuously rises. Whether this is the cause of my random crashing after X minutes of idling I don't yet know. Need to spend more time idling to find out.

MSAA disabled:
![image](https://github.com/user-attachments/assets/a7bc05c7-07e2-4f10-b0c9-bb157351509f)

Enabled:
![image](https://github.com/user-attachments/assets/fcf92633-035d-40fb-b439-36b8bf127777)

### Steps to reproduce

See Description.

### Minimal reproduction project (MRP)

TBD

**Comment by piramidamsc** [comment]
I agree with you, I have the same problem when running my 3D project on opengl 3, from 5-7 minutes and the phone freezes so tightly that I have to restart it. The new redmi note 8t pro is powered by a snapdragon 610 processor. the harder the render, the faster the problem manifests itself. so I had to abandon msaa until better times.
